addons/sale_timesheet/models/sale_order.py

```diff
@@ -226,12 +226,16 @@ def _convert_qty_company_hours(self, dest_company):
     def _timesheet_create_project(self):
         project = super()._timesheet_create_project()
         uom_hour = self.env.ref('uom.product_uom_hour')
-        allocated_hours = 0
+        uom_day = self.env.ref('uom.product_uom_day')
+        uom_unit = self.env.ref('uom.product_uom_unit')
+        allocated_hours = 0.0
         for line in self.order_id.order_line:
             product_type = line.product_id.service_tracking
             if line.is_service and (product_type == 'task_in_project' or product_type == 'project_only'):
-                allocated_hours += line.product_uom_qty * line.product_uom.factor_inv
-        allocated_hours *= uom_hour.factor
+                if line.product_uom == uom_day:
+                    allocated_hours += line.product_uom_qty * line.product_uom.factor_inv * uom_hour.factor
+                elif line.product_uom in [uom_unit, uom_hour]:
+                    allocated_hours += line.product_uom_qty
```

Comment:
Why don't you apply the same changes than the ones in the original PR? 🤔 

The user could also create a new UoM in the same category of `days` and hours UoMs. 🙂

Reply:
> Why don't you apply the same changes than the ones in the original PR? thinking
> 
> The user could also create a new UoM in the same category of `days` and hours UoMs. slightly_smiling_face

Done